dependencies.yaml

```diff
@@ -350,7 +350,7 @@ dependencies:
         packages:
           - cachetools
           - cuda-python>=11.7.1,<12.0
-          - &numba numba>=0.56.4,<0.57
+          - &numba numba>=0.57
```

Comment:
I would recommend removing the actual numba update from this PR and simply merging the rest when we're confident that everything works. We don't even know for sure that numba 0.57 will make it in time for the 23.06 RAPIDS release, and I don't want this PR lingering waiting for that. I'd prefer to get the fixes merged so that the update is painless when we're ready for it.

Reply:
I don't think this PR will pass tests without numba 0.57, because without it numba won't be be set into CEC/MVC mode via the config option.